sdk/tables/data-tables/src/models.ts

```diff
@@ -107,6 +107,10 @@ export interface TableQueryOptions {
  * OData Query options to limit the set of entities returned.
  */
 export interface TableEntityQueryOptions {
+  /**
+   * Maximum number of records to return per page
+   */
+  top?: number;
```

Comment:
Can top be specified by the `maxPageSize` argument to the `byPage` method instead?

Reply:
Yeah you're right. We can set the maxPageSize when listing byPage. It wouldn't really make sense to use it when just iterating over all the items, as it would have no noticeable effect.

I'm going back to not exposing it publicly.

The Docs tool seems to have problems with this definition

```typescript
type InternalListTableEntitiesOptions = ListTableEntitiesOptions & {
  queryOptions?: TableEntityQueryOptions & { top?: number };
};
```

so changing it to its equivalent

```typescript
type InternalQueryOptions = TableEntityQueryOptions & { top?: number };
interface InternalListOptions extends ListTableEntitiesOptions {
  queryOptions?: InternalQueryOptions;
}
```